host/volume/manager/manager.go

```diff
@@ -279,12 +296,12 @@ func (m *Manager) getProviderBucket(tx *bolt.Tx, providerID string) (*bolt.Bucke
 	return providerBucket, err
 }
 
-// Called to sync changes to disk when a VolumeInfo is updated
-func (m *Manager) persistVolume(tx *bolt.Tx, id string) error {
+// Called to sync changes to disk when a volume is updated
+func (m *Manager) persistVolume(tx *bolt.Tx, vol volume.Volume) error {
 	// Save the general volume info
 	volumesBucket := tx.Bucket([]byte("volumes"))
-	k := []byte(id)
-	vol, ok := m.volumes[id]
+	k := []byte(vol.Info().ID)
```

Comment:
somewhere in the area of a vtable jump, but if you're suggesting this reads like a stutter I'd have to concede; will replace.